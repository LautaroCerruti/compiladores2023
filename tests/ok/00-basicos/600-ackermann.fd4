let rec ackermann (m n:Nat) : Nat = 
    ifz m then n + 1 else (
        ifz n then ackermann (m-1) (1) else ackermann (m-1) (ackermann m (n-1))
    )

let ack2350 : Nat = print "ackermann en 2 350 es " (ackermann 2 350)
let ack35 : Nat = print "ackermann en 3 5 es " (ackermann 3 5)
let ack37 : Nat = print "ackermann en 3 7 es " (ackermann 3 7)
let ack40 : Nat = print "ackermann en 4 0 es " (ackermann 4 0)