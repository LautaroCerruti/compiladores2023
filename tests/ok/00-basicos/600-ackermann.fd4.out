ackermann en 2 350 es 703
ackermann en 3 5 es 253
ackermann en 3 7 es 1021
ackermann en 4 0 es 13
